packages/playground/windows/playground/HostingPane.xaml.cpp

```diff
@@ -174,6 +175,9 @@ HostingPane::HostingPane() {
 
   LoadFilenameSettings();
 
+  // Configure paths for custom fonts
+  react::uwp::SetFontFamilyPaths({{L"CustomFont", L"ms-appx:///Assets/seguiemj.ttf#Segoe UI Emoji"}});
```

Comment:
We should think about how we want font auto-linking to work to see if it would affect this API. -- We may want the API to be react::uwp::AddFontFamilyPath(L"CustomFont", L"ms-appx:///Assets/seguiemj.ttf#Segoe UI Emoji") so that the auto-linking could potentially register fonts that are auto-linked by `react-native link`.

Additionally, does this correctly handle all the different font weights.  Often fonts will come as multiple different files for different font weights.  Ex, if SegoeUI was being registered as a custom font you'd need to register `segoeui.ttf` for standard weight, but `segoeuib.ttf` for bold.  But as far the react code both would be using the FontFamily "SegoeUI".   Android added a similar API:  https://github.com/facebook/react-native/blob/796dcf441caf5e2729dd41235568fb3b6a109742/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java#L91 But it reads in an android font resource file, which can specify different font file for different font weights.

Reply:
Barring a change to XAML, different files for font weights will require some juggling within RNW. Despite the XAML FontFamily class representing a collection of typefaces (each a specific combination of family name, weight, and style), the constructor only seems to allow loading of a single local font file.

A TextBlock with FontFamily set to a system font can just change the FontWeight or FontStyle, and the appropriate font file will be selected from the system (e.g. FontFamily=Arial, Style=Italic, Arial Italic is selected). Since a custom FontFamily only links against a single file, any weight and style settings that don't match the file will trigger stylization (e.g. file contains Arial, style = Italic, renders normal Arial glyphs with italic styling).

So to support multiple files for a FontFamily, we'll need to update the FontFamily prop to the correct file any time FontWeight or FontStyle changes, and include some logic for fallbacks.

Reply:
@acoates-ms I added support for registering specific font variants, with fallback logic to use the "Normal" version of a font if an exact match is not found. You now register each variant individually, which should also better match any future `react-native link` needs.